server/config/persist_options.go

```
@@ -598,31 +617,54 @@ func (o *PersistOptions) CheckLabelProperty(typ string, labels []*metapb.StoreLa
 	return false
 }
 
+const ttlConfigPrefix = "/config/ttl/"
+
 // SetTTLData set temporary configuration
-func (o *PersistOptions) SetTTLData(parCtx context.Context, key string, value interface{}, ttl time.Duration) {
-	if data, ok := o.ttl[key]; ok {
-		data.Clear()
-		o.ttlCancel[key]()
+func (o *PersistOptions) SetTTLData(parCtx context.Context, client *clientv3.Client, key string, value string, ttl time.Duration) {
+	if o.ttl == nil {
+		o.ttl = cache.NewStringTTL(parCtx, time.Second*5, time.Minute*5)
 	}
-	ctx, cancel := context.WithCancel(parCtx)
-	o.ttl[key] = cache.NewStringTTL(ctx, 5*time.Second, ttl)
-	o.ttl[key].Put(key, value)
-	o.ttlCancel[key] = cancel
+	kv := clientv3.NewKV(client)
+	grantResp, _ := client.Grant(parCtx, int64(ttl.Seconds()))
+	kv.Put(parCtx, ttlConfigPrefix+key, value, clientv3.WithLease(grantResp.ID))
+	o.ttl.PutWithTTL(key, value, ttl)
 }
 
-func (o *PersistOptions) getTTLData(key string) (interface{}, bool) {
-	if data, ok := o.ttl[key]; ok {
-		return data.Get(key)
+func (o *PersistOptions) getTTLData(key string) (string, bool) {
+	if o.ttl == nil {
+		return "", false
 	}
-	return nil, false
+	if result, ok := o.ttl.Get(key); ok {
+		return result.(string), ok
+	}
+	return "", false
+}
+
+// LoadTTLFromEtcd loads temporary configuration which was persisted into etcd
+func (o *PersistOptions) LoadTTLFromEtcd(ctx context.Context, client *clientv3.Client) error {
+	resps, err := etcdutil.EtcdKVGet(client, ttlConfigPrefix, clientv3.WithPrefix())
+	if err != nil {
+		return err
+	}
+	if o.ttl == nil {
+		o.ttl = cache.NewStringTTL(ctx, time.Second*5, time.Minute*5)
+	}
+	for _, resp := range resps.Kvs {
+		key := string(resp.Key)
+		value := string(resp.Value)
+		leaseID := resp.Lease
+		resp, _ := client.TimeToLive(context.TODO(), clientv3.LeaseID(leaseID))
+		o.ttl.PutWithTTL(key, value, time.Duration(resp.TTL)*time.Second)
+	}
+	return nil
```

I think we shouldn't ignore the error here and why use context.TODO?